### PR TITLE
Ignore Unpitched objects in key analysis

### LINKED ISSUE #1525: An exception occurred: 'Unpitched' object has no attribute 'pitch'
**music21 version**

8.1.0

**Problem summary**
Key detection throws an exception on midi files with drum tracks.

Should be easy to check if track is on channel 10 reserved for MIDI drums and analyse everything but non-percussive parts for overall score key

**Steps to reproduce**

```
    score = music21.converter.parse(filename_mid)
    key = score.analyze(method)

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
Fixes #1525

### CODE REVIEW

**Comment by mscuthbert** [comment]
We should be able to replace this:

```
        for n in streamObj.notes:
            length = n.quarterLength
             if n.isChord:
                 for m in n.pitchClasses:
                     pcDist[m] += length
             elif isinstance(n, percussion.PercussionChord):
                 for m in n.notes:
                     if isinstance(m, note.Note):
                         pcDist[m.pitch.pitchClass] += length
             else:
                 pcDist[n.pitch.pitchClass] += length
```

with:

```
        for n in streamObj.notes:
            length = n.quarterLength
            for p in n.pitches:
                  pcDist[p.pitchClass] += length
```

**Comment by jacobtylerwalls** [comment]
Unfortunately, I see this (and having a vague memory of running into this while dev'ing, should have called it out):

Shall I fix this first in a separate PR?

```python
>>> from music21 import percussion, note
>>> pChord = percussion.PercussionChord([
...     note.Unpitched(),
...     note.Note('E-4'),
...     note.Note('B-4'),
... ])
>>> pChord.notes
(<music21.note.Unpitched object at 0x10d793850>, <music21.note.Note E->, <music21.note.Note B->)
>>> pChord.pitches
()
```

File: music21/analysis/discrete.py [hunk 1]
```diff
@@ -367,11 +368,8 @@ def _getPitchClassDistribution(self, streamObj):
 
         for n in streamObj.notes:
             length = n.quarterLength
-            if n.isChord:
-                for m in n.pitchClasses:
-                    pcDist[m] += length
-            else:
-                pcDist[n.pitch.pitchClass] += length
+            for p in n.pitches:
+                  pcDist[p.pitchClass] += length
```
Comment (jacobtylerwalls) [inline music21/analysis/discrete.py:372]:
```suggestion
                pcDist[p.pitchClass] += length

```

**Comment by mscuthbert** [comment]
THANKS!

**Pull request merged**